### DISCUSSION TITLE
Ability to edit system prompt

### DISCUSSION THREAD

**Discussion by gleitz** [chrislemke/ChatFred#92] [Ideas]
**Is your feature request related to a problem? Please describe.**
I want to be able to set system prompts like I can do on the ChatGPT website. I want this because I would prefer that ChatFred be more precise so, when asking for help with coding tasks, it just gives me the code rather than a long explanation. I also don't want it to apologize or tell me that it is an AI.

**Describe the solution you'd like**
I believe the system prompt could be configurable by making [this prompt](https://github.com/chrislemke/ChatFred/blob/680d414b136e011cc70b757218cf35359042c2bf/workflow/src/text_chat.py#L206) editable by the user.

**Describe alternatives you've considered**
I thought of forking the extension and overwriting that hard-coded prompt myself.

**Additional context**
I know OpenAI uses A system prompt (on the website this is `How would you like ChatGPT to respond?`) and also custom instructions (`What would you like ChatGPT to know about you to provide better responses?`). Not sure what these fields look like in their API.

Here's an example system prompt I would like to use:

```
- Be highly organized

- Suggest solutions that I didn’t think about—be proactive and anticipate my needs

- Treat me as an expert in all subject matter

- Mistakes erode my trust, so be accurate and thorough

- Provide detailed explanations, I’m comfortable with lots of detail

- Value good arguments over authorities, the source is irrelevant

- Consider new technologies and contrarian ideas, not just the conventional wisdom

- You may use high levels of speculation or prediction, just flag it for me

- Recommend only the highest-quality, meticulously designed products like Apple or the Japanese would make—I only want the best

- Recommend products from all over the world, my current location is irrelevant

- No moral lectures

- Discuss safety only when it's crucial and non-obvious

- If your content policy is an issue, provide the closest acceptable response and explain the content policy issue

- Cite sources whenever possible, and include URLs if possible

- List URLs at the end of your response, not inline

- Link directly to products, not company pages

- No need to mention your knowledge cutoff

- No need to disclose you're an AI

- If the quality of your response has been substantially reduced due to my custom instructions, please explain the issue
```

**Comment by gleitz** [comment]
Now that I look at the extension, wondering what the difference between the "jailbreak prompt" and and a system prompt might be.

**Comment by chrislemke** [comment]
Hey @gleitz!
Very good question ;-). 
To be precise, the jailbreak prompt is actually just another user prompt. check out the code:

```
def create_message(prompt: str) -> List[Dict[str, str]]:
    """Creates a message to be sent to the model.

    Args:
        prompt (str): The prompt to be included in the message.

    Returns:
        List[Dict[str, str]]: A list of dictionaries representing the message,
            with each dictionary containing the role and content of the message.
    """
    transformation_pre_prompt = """You are a helpful assistant who interprets every input as raw
    text unless instructed otherwise. Your answers do not include a description unless prompted to do so.
    Also drop any "`" characters from the your response."""

    if __text_transformation_prompt:
        return [
            {"role": "system", "content": transformation_pre_prompt},
            {
                "role": "user",
                "content": f"{__text_transformation_prompt} Don't add any comments: {prompt}",
            },
        ]

    messages = [{"role": "system", "content": "You are a helpful assistant"}]
    for user_text, assistant_text in read_from_log():
        if user_text == __jailbreak_prompt:
            continue
        messages.append({"role": "user", "content": user_text})
        messages.append({"role": "assistant", "content": assistant_text})

    if __jailbreak_prompt and __unlocked == 1:
        messages.append({"role": "user", "content": __jailbreak_prompt})
        messages.append({"role": "assistant", "content": "Okay! How can I help?"})

    messages.append({"role": "user", "content": prompt})
    return messages
```

Feel free to create a pull request in case you want to contribute!

**Comment by April-June-August** [comment]
I want this too! Would be great to switch between several system prompts by just typing something like `cfswitch` to choose from results from Script Filter and add a custom one by adding an argument to it. After which you invoke a typical query with the system prompts.

Would also be great if you can name each prompt (`Code explanation`, `Translate engine`) and show the name as a part of the subtext during a typical query…

I may want to work on this someday :P

**Comment by chrislemke** [comment]
Nice idea. I will move this "issue" into the discussions. @Sponge-bink feel free to create a PR. Thanks in advance.
